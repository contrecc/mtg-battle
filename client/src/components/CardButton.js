import React from 'react';
import { Button } from 'reactstrap';

export default props => {
  const { roundCompleted, card, calculate, id } = props;

  return (
    <Button id={id} className="btn-lg" disabled={roundCompleted} onClick={calculate}>
      {card.name}
    </Button>
  );
};
