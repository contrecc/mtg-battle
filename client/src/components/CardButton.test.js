import React from 'react';
import CardButton from './CardButton';
import { shallow, mount } from 'enzyme';

describe('CardButton', () => {
  let props;

  beforeEach(() => {
    props = {
      roundCompleted: false,
      card: { name: 'Red Wizard' },
      calculate: jest.fn(),
      id: 1
    };
  });

  it('always renders a button', () => {
    const wrapper = mount(<CardButton {...props} />);
    const buttons = wrapper.find('button');

    expect(buttons.length).toBeGreaterThan(0);
  });

  it('should render with props', () => {
    const wrapper = shallow(<CardButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls its onClick function when clicked', () => {
    const wrapper = mount(<CardButton {...props} />);
    const buttons = wrapper.find('button');
    buttons.simulate('click');
    expect(props.calculate).toHaveBeenCalled();
  });
});
