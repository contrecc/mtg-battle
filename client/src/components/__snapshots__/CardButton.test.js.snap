// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton should render with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    calculate={[Function]}
    card={
        Object {
            "name": "Red Wizard",
          }
    }
    id={1}
    roundCompleted={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Red Wizard",
      "className": "btn-lg",
      "color": "secondary",
      "disabled": false,
      "id": 1,
      "onClick": [Function],
      "tag": "button",
    },
    "ref": null,
    "rendered": "Red Wizard",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Red Wizard",
        "className": "btn-lg",
        "color": "secondary",
        "disabled": false,
        "id": 1,
        "onClick": [Function],
        "tag": "button",
      },
      "ref": null,
      "rendered": "Red Wizard",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
